lib/src/level.dart

```diff
@@ -65,7 +65,7 @@ class Level implements Comparable<Level> {
   ];
 
   @override
-  bool operator ==(Object other) => other is Level && value == other.value;
+  bool operator ==(Object? other) => other is Level && value == other.value;
```

Comment:
Why make this nullable? Null should never be passed here.

Reply:
I thought we are always supposed to do that for `==`? Like if you have two `dynamic` or `Object?` objects and check if they are `==` you shouldn't get a type failure.

Reply:
`null` is handled separately. If you do `== null` or `== someNullVar` that value won't ever be passed in. The argument type of this operator on `Object` should also be `Object` and not `Object?`. You _can_ loosen it here to `Object?`, but nothing will ever use that loose time.

Reply:
what kind of black magic is that?